subsys/pm/policy/device_policy_default.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <kernel.h>
+#include <pm/pm.h>
+#include <pm/device_policy.h>
+
+#define LOG_LEVEL CONFIG_PM_LOG_LEVEL /* From power module Kconfig */
+#include <logging/log.h>
+LOG_MODULE_DECLARE(power);
+
+static const enum pm_device_state states_map[] = {
+	PM_DEVICE_STATE_ACTIVE,    /* PM_STATE_ACTIVE */
+	PM_DEVICE_STATE_LOW_POWER, /* PM_STATE_RUNTIME_IDLE */
+	PM_DEVICE_STATE_LOW_POWER, /* PM_STATE_SUSPEND_TO_IDLE */
+	PM_DEVICE_STATE_LOW_POWER, /* PM_STATE_STANDBY */
+	PM_DEVICE_STATE_SUSPENDED, /* PM_STATE_SUSPEND_TO_RAM */
+	PM_DEVICE_STATE_SUSPENDED, /* PM_STATE_SUSPEND_TO_DISK */
+	PM_DEVICE_STATE_OFF,       /* PM_STATE_SOFT_OFF */
+};
```

Comment:
Using this sort of mapping is fragile (a change in the enum order would result in this map to not work as expected without any warning).

Reply:
This just to keep the current behavior. It can be a huge switch / case statement. I don't think the policy will be that simple in the future that is why it was kept this way. And regarding changing, system power sates should not change order frequently  ...
Another possibility is to have an sparse array.

Reply:
A test that makes sure configured mappings are correct would be another option, or maybe build assertions. There needs to be a way to easily catch things like an enum re-ordering, enum field insertion, etc.

Reply:
Changed it. Please take a look again.